File: app/src/main/java/com/michalbaran/incomeapp/service/WalletService.java
package com.michalbaran.incomeapp.service;

import com.michalbaran.incomeapp.database.Expenses;
import com.michalbaran.incomeapp.database.Income;
import com.reactiveandroid.query.Select;

public class WalletService implements IWalletService{
    @Override
    public double totalIncome() {
        return Select.from(Income.class).sum("amount");
    }

    @Override
    public double totalExpenses() {
        return Select.from(Expenses.class).sum("amount");
    }

    @Override
    public double accountBalance() {
        return totalIncome()-totalExpenses();
    }
}

File: app/src/main/java/com/michalbaran/incomeapp/service/WalletService.kt
package com.michalbaran.incomeapp.service

import com.michalbaran.incomeapp.database.Expenses
import com.michalbaran.incomeapp.database.Income
import com.reactiveandroid.query.Select

class WalletService : IWalletService {
    override fun totalIncome(): Double {
        return Select.from<Income>(Income::class.java).sum("amount").toDouble()
    }

    override fun totalExpenses(): Double {
        return Select.from<Expenses>(Expenses::class.java).sum("amount").toDouble()
    }

    override fun accountBalance(): Double {
        return totalIncome() - totalExpenses()
    }
}
